lib/elixir/lib/io/ansi/docs.ex

```diff
@@ -524,7 +524,8 @@ defmodule IO.ANSI.Docs do
   end
 
   defp escape_underlines_in_link(text) do
-    ~r{https?://\S*}
+    # Regular expression adapted from https://tools.ietf.org/html/rfc3986#appendix-B
+    ~r{[a-z][a-z0-9\+\-\.]*://\S*}i
```

Comment:
Just a note, beca use +-. are actually reserved characters, except inside `[` and `]`, I would rather train the eye to always escape them, which is why I have added the escapes here. Removing the escape for `/` was a great call though.